src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -351,12 +402,15 @@ BuildDistinctPlan(Query *masterQuery, Plan *subPlan)
 		column->varno = OUTER_VAR;
 	}
 
+	containstDistinctAggregate = HasDistinctAggregate(masterQuery);
+
 	/*
 	 * Create group by plan with HashAggregate if all distinct
-	 * members are hashable, Otherwise create sort+unique plan.
+	 * members are hashable, and not containing distinct aggregate.
+	 * Otherwise create sort+unique plan.
 	 */
 	distinctClausesHashable = grouping_is_hashable(distinctClauseList);
-	if (distinctClausesHashable)
+	if (distinctClausesHashable && !containstDistinctAggregate)
```

Comment:
I think we can rename `containstDistinctAggregate` to `hasDistinctAggregate` in order to be consistent.

Visual. I would move `distinctClausesHashable = grouping_is_hashable(distinctClauseList);` below `HasDistinctAggregate ()` above.

Reply:
ok